File: 0x00-hello_world/0-preprocessor.go
package main

import (
	"log"
	"os"
	"os/exec"
)

/*
Run the C preprocessor on a C file and output
the result to a file named "c".

Returns:
	Exit code 0 on success.
*/

func main() {
	cfile := os.Getenv("CFILE")

	cmd := excec.Command("gcc", "-E", cfile, "-o", "c")

	if err := cmd.Run(); err != nil {
		log.Fatal(err)
	}
}

File: 0x00-hello_world/0-preprocessor.js
#!/usr/bin/env node

/**
 * Run the gcc preprocessor on a C file and output the
 * result to a file named "c".
 */

const { execFileSync } = require('child_process');

/**
 * Program entry point.
 * 
 * @returns{number} Process exit code.
 */
const cfile = process.env.CFILE;

execFileSync(
    "gcc", ["-E", cfile, "-o", "c"], {stdio: "inherit",}
);